src/billing/components/Free/OrgLimits.tsx

```diff
@@ -1,78 +1,110 @@
+// Libraries
 import React, {FC} from 'react'
 import {useSelector} from 'react-redux'
-import {Grid, Columns} from '@influxdata/clockface'
+import {Columns, Grid} from '@influxdata/clockface'
+
+// Components
 import OrgLimitStat from 'src/billing/components/Free/OrgLimitStat'
 
+// Selectors
+import {isOrgIOx} from 'src/organizations/selectors'
+
 // Types
 import {AppState} from 'src/types'
 import {Limit} from 'src/cloud/actions/limits'
 import {LimitsState} from 'src/cloud/reducers/limits'
 
-const limits = (orgLimits: LimitsState | {}) =>
+interface LimitCardProps {
+  limitName: string
+  limitValue: Limit
+}
+
+const removeUnrenderedProps = (orgLimits: LimitsState | {}) =>
   Object.entries(orgLimits).filter(
     ([limitName]) =>
       limitName !== 'orgID' &&
       limitName !== 'status' &&
       limitName !== 'endpoints'
   )
 
-const OrgLimits: FC = () => {
+const LimitCard: FC<LimitCardProps> = ({limitName, limitValue}) => {
+  return (
+    <Grid.Column
+      key={limitName}
+      widthXS={Columns.Twelve}
+      widthSM={Columns.Six}
+      widthMD={Columns.Four}
+      widthLG={Columns.Three}
+    >
+      <OrgLimitStat name={limitName} value={limitValue.maxAllowed} />
+    </Grid.Column>
+  )
+}
+
+export const OrgLimits: FC = () => {
   const orgLimits = useSelector((state: AppState) => state.cloud?.limits ?? {})
+  const orgUsesIOx = useSelector(isOrgIOx)
 
   return (
     <Grid>
-      {limits(orgLimits).flatMap(([name, value]: [string, Limit]) => {
-        if (name === 'buckets') {
-          return Object.entries(value)
-            .filter(([n]) => n !== 'limitStatus')
-            .map(([n, v]: [string, number]) => {
+      {removeUnrenderedProps(orgLimits).map(limit => {
+        const limitName = limit[0]
+        const limitValue = limit[1]
+
+        // The "Rate" category includes multiple object literal 'limits', each with its own 'limitStatus.'
+        if (limitName === 'rate') {
```

Comment:
only real piece of feedback I have is maybe `limitName` should be based off a type or an enum, rather than a string. Do any parts of `openapi` define these?

Reply:
There doesn't appear to be a satisfying way to do this, unfortunately: `openapi` does define an interface `Limit`, but not an enum. The slick solution would be converting the keys of the _interface_ in openapi into an enum. I can't find a way to do it in the Typescript docs though, and [I think this is why: ](https://stackoverflow.com/questions/72359790/typescript-generate-enum-from-an-interface-keys-with-index-number)

> actually now that I think about it, I think it is not possible to convert an interface to an enum at compile since enums are a runtime thing and without external tools this might be impossible. The other way around might work though. Is enum to interface something that would fit your needs? – 
> [Tobias S.](https://stackoverflow.com/users/8613630/tobias-s)
> 
>  [May 24, 2022 at 8:54](https://stackoverflow.com/questions/72359790/typescript-generate-enum-from-an-interface-keys-with-index-number#comment127832528_72359790)
> 
> I don't see how that would work. Types only exist at compile time, but you need the values of the enum at runtime. – 
> [Robby Cornelissen](https://stackoverflow.com/users/3558960/robby-cornelissen)
>  [May 24, 2022 at 8:55](https://stackoverflow.com/questions/72359790/typescript-generate-enum-from-an-interface-keys-with-index-number#comment127832534_72359790) 
> Not really possible. The order is problematic to begin with (as you mention it's important)

Reply:
I'll write out an enum anyway - that's likely to be less buggy here so that someone doesn't fatfinger one of the names and cause one of the limits to be missing.